PWGHF/vertexingHF/AliAnalysisTaskHFSimpleVertices.cxx

```diff
@@ -1267,10 +1479,15 @@ Int_t AliAnalysisTaskHFSimpleVertices::SelectInvMassAndPtDzero(TObjArray* trkArr
   Double_t hilim=fDzeroSkimCuts[2];
   pdg2[0]=211; pdg2[1]=321;   // pi+ K- --> D0
   minv2 = rd4massCalc2->InvMass2(nprongs,pdg2);
-  if(minv2>lolim*lolim && minv2<hilim*hilim ) retval+=1;
+  if(minv2>lolim*lolim && minv2<hilim*hilim ) retval += (1 << kbitDzero);
   pdg2[0]=321; pdg2[1]=211;   // K+ pi- --> D0bar
   minv2 = rd4massCalc2->InvMass2(nprongs,pdg2);
-  if(minv2>lolim*lolim && minv2<hilim*hilim ) retval+=2;
+  if(minv2>lolim*lolim && minv2<hilim*hilim && (retval & (1 << kbitDzero)) == 0) retval+=(1 << kbitDzero);
```

Comment:
This line will create problems for the D0 because it breaks the logic used to tag separately D0 and D0bar candidates. A candidate (+- pair) can be OK as D0 (which is what is tested at L1482), D0bar (tested here) or both. In case it is OK only as D0 we should return 1, and this is done correctly in the new code. In case it is OK for both we should return 3, and this is also OK in the new code. What does not work is the case in which it is OK only as D0ber, in which we should return 2, and instead the new code returns 1.
The simplest way would be to redefine the selection bits as kbitDzero, kbitDzerobar and kbitJpsi.

Reply:
Thanks a lot for comments, I will make a commit to redefine the selection bits (add kbitDzerobar).